src/runtime_src/core/include/ert.h

```diff
@@ -146,6 +146,19 @@ struct ert_start_kernel_cmd {
   uint32_t data[1];            /* count-1 number of words */
 };
 
+/**
+ * struct ert_dpu_data - interpretation of data payload for ERT_START_DPU
+ *
+ * The data payload for ERT_START_DPU is interpreted as fixed instruction
+ * buffer address along with instruction count, followed by regular kernel
+ * arguments.
+ */
+struct ert_dpu_data {
+  uint64_t instruction_buffer;  /* buffer address 2 words */
+  uint32_t instruction_count;   /* number of instructions */
```

Comment:
is this really number of instructions? if so, then how big is one instruction?

Reply:
No clue.    For raw ptr it was defined as number of instruction words and passed as kernel argument.  I actually think we should make it the size of the buffer object and leave it to KMD to convert to whatever firmware needs.

Reply:
Size of the buffer in bytes sounds good :-).